src/puppetlabs/puppetdb/cli/time_shift_export.clj

```diff
@@ -76,13 +78,48 @@
       "facts" (time-shift-producer-timestamp json-data shift-period)
       "export-metadata" json-data)))
 
+(defn find-command-version
+  [entity]
+  (let[latest-command-versions command-constants/latest-command-versions]
+    (case entity
+      "facts" (:replace_facts latest-command-versions)
+      "catalogs"(:replace_catalog latest-command-versions)
+      "reports" (:store_report latest-command-versions))))
+
+(def counter (atom -1))
+
+(defn next-index []
+  (swap! counter inc))
+
+(defn rename-filenames
```

Comment:
rename-filename, because you change only one

Reply:
Updated